toxcore/tox.c

```diff
@@ -2193,7 +2225,26 @@ bool tox_friend_send_lossy_packet(Tox *tox, uint32_t friend_number, const uint8_
 
 void tox_callback_friend_lossy_packet(Tox *tox, tox_friend_lossy_packet_cb *callback)
 {
-    tox->friend_lossy_packet_callback = callback;
+    /** start at PACKET_ID_RANGE_LOSSY_CUSTOM_START so ToxAV Packets are excluded if the old setter function is used */
+    for (uint8_t i = PACKET_ID_RANGE_LOSSY_CUSTOM_START; i <= PACKET_ID_RANGE_LOSSY_END; ++i) {
+        tox->friend_lossy_packet_callback_per_pktid[i] = callback;
+    }
+}
+
+void tox_callback_friend_lossy_packet_per_pktid(Tox *tox, tox_friend_lossy_packet_cb *callback, uint8_t pktid)
+{
+    // no error code returned in this function?
```

Comment:
If you want, you can make it return an error. It's internal (toxav) only. Maybe log it somewhere. Maybe add an assert, because if this function errors, we have a code bug.

Reply:
no